.github/changelog-processor.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import argparse
+
+parser = argparse.ArgumentParser(description="Process the CHANGELOG.md")
+parser.add_argument(
+    "changelog",
+    metavar="CHANGELOG",
+    help="Path to the CHANGELOG.md file"
+)
+parser.add_argument(
+    "--print-latest-version",
+    dest="print_latest_version",
+    help="Print the latest version (first in the file) found in the CHANGELOG.md",
+    action="store_true"
+)
+parser.add_argument(
+    "--should-release",
+    dest="should_release",
+    help="Should a release be made? Prints `1` or `0`.",
+    action="store_true"
+)
+
+args = parser.parse_args()
+
+with open(args.changelog, "r") as changelog:
+    lines = changelog.readlines()
+
+    # Find the latest version
+    for line in lines:
+        if not line.startswith("## ["):
+            continue
+
+        version = line.strip().removeprefix("## [").split("]")[0]
+        break
+
+    if args.print_latest_version:
+        print(version, end = "")
+        sys.exit(1)
```

Comment:
Why does it error in this case?

Reply:
My brain wasn't working :P I had forgotten that `0` is "success"